client/src/components/LogInModal.jsx

```diff
@@ -0,0 +1,94 @@
+//Import necessary functionalities
+import React from 'react';
+import { UserStatus } from '../context/UserContext.jsx';
+import { useNavigate } from 'react-router-dom';
+
+const LogInModal = ({ isOpen, onClose }) => {
+
+    if (!isOpen) return null;
+
+    //Import display and user settings to use in display control and user updates
+    const { loggedUser, setLoggedUser } = UserStatus();
+    
+    //Call useNavigate hook to change component display upon log in
+    const navigate = useNavigate();
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+        
+        //Check that correct value is being submitted right now
+        console.log(event.currentTarget.username.value);
+
+        //Set the value of the username input field to be stored in username variable
+        const username = event.currentTarget.username.value;
+
+        //Future Plans: Add in password functionality - OAuth?
+
+        //If on submission, there is a username submitted, send it to the server to check its existence
+        if (username) {
+            try {
+                const response = await fetch('/api/users/', {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify({ username })
+                });
+
+                if (!response.ok) {
+                    throw new Error("Failed to check log in");
+                }
+
+                const userStatus = await response.json();
+
+                //If there is a user with this username, log them in
+                if (userStatus.exists) {
+
+                    //Switch to dashboard display
+                    navigate('/dashboard');
+
+                    //Set the logged user state to the user id we will need to get the correct data
+                    setLoggedUser({userId: userStatus.userId, name: userStatus.name});
+                    
+                    //Potentially use to combat losing user on refresh:
+                    // localStorage.setItem("user", userStatus.userId);
+
+                    //Close the modal
+                    onClose();
+                } else {
+                    //Client side error handling to ask them to try again - need to update once log in is fleshed out
+                    console.log("Username does not exist.");
+                }
+            } catch (error) {
+                //Update this error handling as well once log in is complete
+                console.error({ message: "Error checking user name", details: error });
+            }
+        } else {
+            //Placeholder alert for temporary error handling - will implement robust form error handling while building out full log in component in Week 3
+            alert("Please make sure to enter a username.");
```

Comment:
When you do your error handling, think about field errors and form errors.  Form errors usually show up at the top of the page (for example Login Failed). Field errors usually make the field component red and have an error message right next to the field that errored (for ex. Please enter a username for a missing username).  Something to keep in mind when you get there.

Reply:
Though the log-in component will be implemented Week 4, this is also a great point for the form submission in the Session Form component. I'll add both types to that while tackling [SCRUM-107: Display Error Handling](https://tatig.atlassian.net/browse/SCRUM-107), our subtask for the form, and keep that in mind when I reach the log in part next week. Thank you!